text/stacks/0004-jammy-jellyfish.md

```diff
@@ -0,0 +1,106 @@
+of March 2023. In order to ensure that Paketo continues to provide a supported
+Ubuntu-based stack for our users, we should produce a new set of stack images,
+based on Jammy, alongside the current Bionic stacks.
+
+## Detailed Explanation
+
+### Variants
+
+The Jammy stack will be delivered in 3 variants that align to those already
+offered in the Bionic stack. The variants will be called `base`, `full`, and
+`tiny` just as they are in the Bionic stack. Each should be developed to offer
+roughly the same set of OS-level package support as their Bionic equivalent.
+
+### Stack IDs
+
+Each of these stacks will declare their stack ID as `io.paketo.stack.jammy`.
+This choice differs from Bionic in that we will **NOT** differentiate between
+the `tiny` and `base/full` variants using the stack ID.
+
+### User IDs
+
+These stacks will differ from Bionic with regards to their UID definitions.
+Each variant will ensure that the UID for the `build` phase is different than
+the `run` phase. This change will align more closely with the
+[recommendations](https://github.com/buildpacks/rfcs/blob/main/text/0085-run-uid.md)
+outlined in the Buildpacks Specification.
+
+### Image Naming and Tagging
+
+The stacks will name and tag their release images with the following pattern:
+
+```
+paketobuildpacks/jammy-{phase}-{variant}:{version}
+```
+
+For example we could see the following images for Jammy stacks:
+
+* `paketobuildpacks/jammy-build-base:latest`
+* `paketobuildpacks/jammy-run-tiny:1.2.3`
+
+This choice is different than the choice made with Bionic to better align the
+image repository references with their logical stack definition. This means
+that, with this naming scheme, it will be much more reasonable to understand
+what is stored at the repository reference
+`paketobuildpacks/jammy-run-base:latest`.
+
+### SBOM
+
+This stack will **NOT** include an SBOM of the form outlined in Stacks
+[RFC0001](https://github.com/paketo-buildpacks/rfcs/blob/main/text/stacks/0001-stack-package-metadata.md).
+This choice is to ready ourselves to adopt whatever Stack SBOM functionality is
+specified by the Cloud Native Buildpacks project while ensuring we don't
+perpetuate the use of our own format for the entirety of this stack's support
+lifetime.
+
+This stack will adopt whatever official SBOM support is declared upstream by
+the CNB project.
+
+## Rationale and Alternatives
+
+We should at least discuss what it would mean to provide a Debian stack. There
+are [LTS releases](https://wiki.debian.org/LTS) for Debian, but they follow a
+different lifecycle and may not provide enough overlap to allow end-users to
+migrate completely.
+
+## Implementation
+
+After the [final
+release](https://discourse.ubuntu.com/t/jammy-jellyfish-release-schedule/23906)
+date on April 21, 2022, we can release our official stack versions.
+
+### Repositories
+
+The Stacks subteam will create 3 new repos for Jammy:
+
+* `jammy-base-stack`
+* `jammy-full-stack`
+* `jammy-tiny-stack`
```

Comment:
The repos for the current stacks are `<full/base/tiny>-stack-release`. Could we follow suit and name the repositories `jammy-<full/base/tiny>-stack-release` for consistency?

Reply:
Why do these repos have the `-release` suffix on them? I've never really understood that part.

Reply:
Not sure, I didn't make them. Probably to show that that's the repository where releases are cut. Most of the heavy duty logic and important stuff, like the `create-stack`, Dockerfiles, and package lists live in the Stacks repo, so all that goes down in the `-release` repos are just for putting the stack together and releasing it. They don't have much else.

The `-release` suffix doesn't release bother me and as above I'd rather everything just be consistent to make our lives every so slightly  easier

Reply:
I'd like to see us move the stack metadata into these variant-specific repos. Its weird that we create a stack in one repo and then release it in another, no?

Especially if the Stack descriptor RFC is adopted, then the utility of the `stacks` repo starts to come into question.

Reply:
Just offering an outsider's point of view on the different repos, they are confusing. It takes some effort to track down where things live and what is located in each part. I think this comes into play if you want to see what's in the various images, or fork and customize. 

It's also a problem if you want to open an issue. Where would one go? It's not exactly clear without investigating more, and not all users will put in that level of effort.

Again, my outsider's point of view would be to simplify and consolidate repos (across stacks & builders). I recognize that may not be technically practical, which is fine, but I think we have to own the documentation and explanation of this in our repos then. [What's here](https://paketo.io/docs/concepts/stacks/#what-is-a-stack) is a start, but it's missing docs around the repos, how to navigate them, and an explanation of how the systems come together to produce the stack.

Reply:
I think the biggest issue we encountered with a consolidated repository structure was in releasing the sub-artifacts (stack or builder) on differing cadences. Tracking things like "what is the next version I should release?" become harder when the repo contains more than one logical artifact. I'm all for documenting this better, but the consolidation problem is probably a bigger issue than I think we should outline a solution for in this RFC.

Reply:
I'm OK with that, taking the documentation approach. This is an area where I've heard feedback, so I wanted to make sure it's on the record. Thanks

Reply:
In keeping with the resolutions above, I'm OK to do this for jammy stacks, and then we can open a subsequent RFC down the line to move the bionic stacks from the current locations to repositories of naming convention `bionic-<stack>-stack`, as well as propose other changes to the bionic stacks to align with decisions here.